pants_plugins/sendwave/pants_docker/package.py

```diff
@@ -223,16 +220,36 @@ async def package_into_image(
     process_path = docker_paths.first_path.path
     # build an list of arguments of the form ["-t",
     # "registry/name:tag"] to pass to the docker executable
+    full_target_name = _build_full_target_name(
+        target_name, field_set.registry.value
+    )
     tag_arguments = _build_tag_argument_list(
-        target_name, field_set.tags.value or [], field_set.registry.value
+        full_target_name, field_set.tags.value or []
     )
     # create the image
-    process_args = [process_path, "build"]
+    # We need to enable BuildKit if we plan to read and write to a remote cache
+    cli_command = (
+        ["buildx", "build"] if docker.options.cache_from else ["build"]
+    )
+    process_args = [process_path] + cli_command
```

Comment:
```suggestion
    process_args = [process_path, cli_command]
```

Reply:
Hm, I don't see why this would be necessary? We want to create a list where `process_path` is the first element and then extend that with the list represented by `cli_command`, so wouldn't the suggested snippet produce e.g. `["path/to/docker", ["buildx", "build"]]`?

Reply:
oops! nvm!!!